arangod/ClusterEngine/ClusterCollection.cpp

```diff
@@ -64,31 +63,7 @@ ClusterCollection::ClusterCollection(LogicalCollection& collection, ClusterEngin
       _engineType(engineType),
       _info(info),
       _selectivityEstimates(collection) {
-  // duplicate all the error handling
-  if (_engineType == ClusterEngineType::MMFilesEngine) {
-    bool isVolatile = Helper::getBooleanValue(_info.slice(), "isVolatile", false);
-
-    if (isVolatile && _logicalCollection.waitForSync()) {
-      // Illegal collection configuration
-      THROW_ARANGO_EXCEPTION_MESSAGE(
-          TRI_ERROR_BAD_PARAMETER,
-          "volatile collections do not support the waitForSync option");
-    }
-
-    VPackSlice journalSlice = _info.slice().get("journalSize");
-
-    if (journalSlice.isNone()) {
-      // In some APIs maximalSize is allowed instead
-      journalSlice = _info.slice().get("maximalSize");
-    }
-
-    if (journalSlice.isNumber()) {
-      if (journalSlice.getNumericValue<uint32_t>() < TRI_JOURNAL_MINIMAL_SIZE) {
-        THROW_ARANGO_EXCEPTION_MESSAGE(TRI_ERROR_BAD_PARAMETER,
-                                       "<properties>.journalSize too small");
-      }
-    }
-  } else if (_engineType == ClusterEngineType::RocksDBEngine) {
+  if (_engineType == ClusterEngineType::RocksDBEngine) {
```

Comment:
shouldn't we use `assert` instead?

Reply:
There is still the MockEngine, so we need at least one `if`.